ext/cld3/extconf.rb

```diff
@@ -17,7 +17,9 @@
 require "mkmf"
 
 # Check pkg-config first to inform the library is missing if so.
-pkg_config("protobuf")
+unless pkg_config("protobuf") || have_library("protobuf")
```

Comment:
`have_library` should be removed because it ignores `pkg-config` tells. `have_library` is not documented well, probably because it could be environment-dependent, but it actually just adds `-lprotobuf` and tries to compile. See
https://github.com/ruby/ruby/blob/7d4cdab8f8c7e1f3f7e5c2add712659057f643fc/lib/mkmf.rb#L927.
Reading `pkg-config` is more reliable than appending `-lprotobuf`, which could miss additional flags required for compiling and linking, and that is why `pkg-config` exists.